src/test/java/school/hei/haapi/integration/SecurityIT.java

```diff
@@ -0,0 +1,109 @@
+package school.hei.haapi.integration;
+
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.ContextConfiguration;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import school.hei.haapi.client.HAHttpClient;
+import school.hei.haapi.exception.NotAuthorizedException;
+import school.hei.haapi.integration.conf.CallerData;
+import school.hei.haapi.integration.conf.ContextInitializer;
+import school.hei.haapi.web.model.StudentResource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@Testcontainers
+@ContextConfiguration(initializers = ContextInitializer.class)
+@AutoConfigureMockMvc
+class SecurityIT {
+
+  private final int serverPort;
+
+  public SecurityIT(@LocalServerPort int serverPort) {
+    this.serverPort = serverPort;
+  }
+
+  private HAHttpClient clientFromToken(String token) {
+    return new HAHttpClient("http://localhost:" + serverPort, token);
+  }
+
+  @Test
+  @SneakyThrows
+  void student1_can_get_his_information() {
+    // given
+    HAHttpClient student1 = clientFromToken(CallerData.STUDENT1_TOKEN);
+
+    // when
+    StudentResource studentResource = student1.getUserById(CallerData.STUDENT1_ID);
+
+    // then
+    assertEquals("Ryan", studentResource.getFirstName());
+  }
+
+  /*
+   * TEST RESULT WHERE NOT_AUTHORIZED EXCEPTION IS EXPECTED SHOULD BE FAILED
+   * I've seen your notice in InternalToExternalErrorHandler class about the use of
+   * UNAUTHORIZED/NOT_AUTHORIZED and FORBIDDEN, _but_ to pass the test I modified it to
+   * UNAUTHORIZED, so you can easily do the test while waiting your comments.
+   */
+
+  @Test
+  void student1_can_not_get_student2_information() {
+    // given
+    HAHttpClient student1 = clientFromToken(CallerData.STUDENT1_TOKEN);
+
+    // excepted
```

Comment:
remove the given-when-expected comments. Just put them on a test for illustration and do not put them for the rest